src/OrchardCore/OrchardCore.ReCaptcha.Core/TagHelpers/ReCaptchaTagHelper.cs

```diff
@@ -49,57 +29,23 @@ public ReCaptchaTagHelper(IOptions<ReCaptchaSettings> optionsAccessor, IResource
         [HtmlAttributeName("language")]
         public string Language { get; set; }
 
-        public override async Task ProcessAsync(TagHelperContext context, TagHelperOutput output)
+        public override Task ProcessAsync(TagHelperContext context, TagHelperOutput output)
         {
-            var robotDetectors = _httpContextAccessor.HttpContext.RequestServices.GetServices<IDetectRobots>();
-            var robotDetected = robotDetectors.Invoke(d => d.DetectRobot(), _logger).Any(d => d.IsRobot) && Mode == ReCaptchaMode.PreventRobots;
-            var alwaysShow = Mode == ReCaptchaMode.AlwaysShow;
-            var isConfigured = _settings != null;
-
-            if (isConfigured && (robotDetected || alwaysShow))
+            void RenderDivToTagHelper(TagBuilder builder)
             {
-                await ShowCaptcha(output);
+                output.TagName = builder.TagName;
+                output.MergeAttributes(builder);
+                output.TagMode = TagMode.StartTagAndEndTag;
             }
-            else
+
+            void DoNotRenderTagHelper()
             {
                 output.SuppressOutput();
             }
-        }
 
-        private async Task ShowCaptcha(TagHelperOutput output)
-        {
-            output.TagName = "div";
-            output.Attributes.SetAttribute("class", "g-recaptcha");
-            output.Attributes.SetAttribute("data-sitekey", _settings.SiteKey);
-            output.TagMode = TagMode.StartTagAndEndTag;
-
-            var builder = new TagBuilder("script");
-            var cultureInfo = await GetCultureAsync();
-
-            var settingsUrl = $"{_settings.ReCaptchaScriptUri}?hl={cultureInfo.TwoLetterISOLanguageName}";
-
-            builder.Attributes.Add("src", settingsUrl);
-            _resourceManager.RegisterFootScript(builder);
-        }
-
-        private async Task<CultureInfo> GetCultureAsync()
-        {
-            var language = Language;
-            CultureInfo culture = null;
-
-            if (string.IsNullOrWhiteSpace(language))
-                language = await _localizationService.GetDefaultCultureAsync();
-
-            try
-            {
-                culture = CultureInfo.GetCultureInfo(language);
-            }
-            catch (CultureNotFoundException)
-            {
-                _logger.LogWarning(S["Language with name {0} not found", language]);
-            }
+            _reCaptchaService.ShowCaptchaOrCallCalback(Mode, Language, RenderDivToTagHelper, DoNotRenderTagHelper);
```

Comment:
Could we use the function directly as lambda expression instead

Reply:
Are you about `DoNotRenderTagHelper`?